mne/viz/ica.py

```diff
@@ -206,19 +240,49 @@ def _plot_ica_sources_evoked(evoked, exclude, title, show):
     times = evoked.times * 1e3
 
     # plot unclassified sources
-    plt.plot(times, evoked.data.T, 'k')
-    for ii in exclude:
-        # use indexing to expose event related sources
-        plt.plot(times, evoked.data[ii].T, color='r', label='ICA %02d' % ii)
+    lines = list()
+    texts = list()
+    labels = list()
+    if picks is None:
+        picks = np.arange(evoked.data.shape[0])
+    # plt.plot(times, evoked.data[picks].T, 'k')
+    for ii in picks:
+        if ii in exclude:
+            label = 'ICA %02d' % ii
```

Comment:
No does not make sense @agramfort the ica.ch_names are the data channel names. This is why we have this in the original code.